File: ios/XiriGoEcommerce/Core/DesignSystem/Component/XGCategoryIcon.swift
import SwiftUI

// MARK: - XGCategoryIcon

/// A colored rounded-rectangle tile with an icon and label below.
///
/// Token source: `shared/design-tokens/components/atoms/xg-category-icon.json`.
///
/// Token mapping:
/// - tileSize      = 79pt
/// - cornerRadius  = `$foundations/spacing.cornerRadius.medium` -> `XGCornerRadius.medium` (10pt)
/// - iconSize      = 40pt
/// - iconColor     = `$foundations/colors.light.iconOnDark`     -> `XGColors.iconOnDark`
/// - labelFont     = `$foundations/typography.typeScale.captionMedium` -> `XGTypography.captionMedium` (12pt Medium)
/// - labelColor    = `$foundations/colors.light.textPrimary`    -> `XGColors.onSurface`
/// - labelSpacing  = 6pt
struct XGCategoryIcon: View {
    // MARK: - Lifecycle

    init(
        name: String,
        systemIconName: String,
        backgroundColor: Color,
        action: @escaping () -> Void,
    ) {
        self.name = name
        self.systemIconName = systemIconName
        self.backgroundColor = backgroundColor
        self.action = action
    }

    // MARK: - Internal

    var body: some View {
        Button(action: action) {
            VStack(spacing: Constants.labelSpacing) {
                ZStack {
                    RoundedRectangle(cornerRadius: XGCornerRadius.medium)
                        .fill(backgroundColor)
                        .frame(
                            width: Constants.tileSize,
                            height: Constants.tileSize,
                        )

                    Image(systemName: systemIconName)
                        .font(.system(size: Constants.iconSize))
                        .foregroundStyle(XGColors.iconOnDark)
                        .accessibilityHidden(true)
                }

                Text(name)
                    .font(XGTypography.captionMedium)
                    .foregroundStyle(XGColors.onSurface)
                    .lineLimit(1)
            }
            .frame(width: Constants.tileSize)
        }
        .buttonStyle(.plain)
        .accessibilityLabel(name)
    }

    // MARK: - Private

    private enum Constants {
        static let tileSize: CGFloat = 79
        static let iconSize: CGFloat = 40
        static let labelSpacing: CGFloat = 6
    }

    private let name: String
    private let systemIconName: String
    private let backgroundColor: Color
    private let action: () -> Void
}

// MARK: - Previews

#Preview("XGCategoryIcon") {
    HStack(spacing: XGSpacing.base) {
        XGCategoryIcon(
            name: "Electronics",
            systemIconName: "desktopcomputer",
            backgroundColor: XGColors.categoryBlue,
            action: {},
        )
        XGCategoryIcon(
            name: "Fashion",
            systemIconName: "tshirt",
            backgroundColor: XGColors.categoryPink,
            action: {},
        )
        XGCategoryIcon(
            name: "Home",
            systemIconName: "house",
            backgroundColor: XGColors.categoryYellow,
            action: {},
        )
    }
    .padding()
    .xgTheme()
}

File: android/app/src/main/java/com/xirigo/ecommerce/core/designsystem/component/XGCategoryIcon.kt
package com.xirigo.ecommerce.core.designsystem.component

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Devices
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.xirigo.ecommerce.core.designsystem.theme.XGColors
import com.xirigo.ecommerce.core.designsystem.theme.XGCornerRadius
import com.xirigo.ecommerce.core.designsystem.theme.XGTheme

// Token source: shared/design-tokens/components/atoms/xg-category-icon.json
// tileSize      = 79dp
// cornerRadius  = $foundations/spacing.cornerRadius.medium → XGCornerRadius.Medium (10dp)
// iconSize      = 40dp
// iconColor     = $foundations/colors.light.iconOnDark     → XGColors.IconOnDark
// labelFont     = $foundations/typography.typeScale.captionMedium → labelMedium (12sp Medium)
// labelColor    = $foundations/colors.light.textPrimary     → XGColors.OnSurface
// labelSpacing  = 6dp
private val TileSize = 79.dp
private val IconSize = 40.dp
private val LabelSpacing = 6.dp

/** Category icon tile with colored background, icon, and label text. */
@Composable
fun XGCategoryIcon(
    name: String,
    icon: ImageVector,
    backgroundColor: androidx.compose.ui.graphics.Color,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .widthIn(max = TileSize)
            .semantics(mergeDescendants = true) {}
            .clickable(onClick = onClick),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Box(
            modifier = Modifier
                .size(TileSize)
                .clip(RoundedCornerShape(XGCornerRadius.Medium))
                .background(backgroundColor),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                modifier = Modifier.size(IconSize),
                tint = XGColors.IconOnDark,
            )
        }
        Spacer(modifier = Modifier.height(LabelSpacing))
        Text(
            text = name,
            style = MaterialTheme.typography.labelMedium,
            color = XGColors.OnSurface,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis,
        )
    }
}

@Preview(showBackground = true)
@Composable
private fun XGCategoryIconPreview() {
    XGTheme {
        XGCategoryIcon(
            name = "Electronics",
            icon = Icons.Outlined.Devices,
            backgroundColor = XGColors.CategoryBlue,
            onClick = {},
        )
    }
}
